bin/node/runtime/src/lib.rs

```diff
@@ -1835,6 +1836,15 @@ impl_runtime_apis! {
 		}
 	}
 
+	impl pallet_nomination_pools_rpc_runtime_api::NominationPoolsApi<Block, AccountId, Balance> for Runtime {
```

Comment:
To be fair, I mainly wanted to emphasize that you cannot unwrap here. Adding a custom error and all, I am not 100% sure about. All runtime APIs are wrapped in a `Result` by default. So essentially, you are now returning a `Result<Result<_, _>, _>`. I think it is best to remove `NpApiError`, and just handle errors by collapsing to `default`, i.e. `unwrap_or_default()`

Sorry for the confusion, I should have clarified this better.

Reply:
Ok, I agree that it is not the cleanest to have a Result wrapped in a Result, it just causes confusion.